helix-core/src/main/java/org/apache/helix/tools/ClusterSetup.java

```diff
@@ -176,6 +179,37 @@ public void close() {
     }
   }
 
+  public void addCluster(String clusterName, boolean overwritePrevious, boolean isCloudEnabled,
+      CloudConfig cloudConfig) throws HelixException {
+    if (isCloudEnabled) {
+      if (cloudConfig != null) {
+        String cloudID = cloudConfig.getCloudID();
+        String cloudProvider = cloudConfig.getCloudProvider();
+        List<String> cloudInfoSource = cloudConfig.getCloudInfoSources();
+        String cloudProcessorName = cloudConfig.getCloudInfoProcessorName();
+
+        // Validate the cloudConfig in the builder
+        // If it is invalid, do not even create a cluster
+        CloudConfig.Builder builder = new CloudConfig.Builder(clusterName);
+        builder.setCloudEnabled(true);
```

Comment:
It's better to use chained builder so that you only need to have one line of code. I noticed some existing ones don't use chain. But for concise code, let's use it.

Reply:
Done.